### PR TITLE
fix: table fixed column borders fix, table improvements

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue

Closes [Fundamental NGX #5097](https://github.com/SAP/fundamental-ngx/issues/5097).

## Description

Table:
* Fixed columns border styles fix.
* Condensed mode selection checkbox styles fix.
* Documentation title fix.

### Before:
1.
Screenshot 2021-07-06 at 21 00 13
2.
Screenshot 2021-07-06 at 21 03 46
3.
Screenshot 2021-07-06 at 21 06 11

### After:
1.
Screenshot 2021-07-06 at 21 03 21
2.
Screenshot 2021-07-06 at 21 05 11
3.
Screenshot 2021-07-06 at 21 06 49

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [X] All values are in `rem`
- [X] Text elements follow the truncation rules
- [X] hover state of the element follow design spec
- [X] focus state of the element follow design spec
- [X] active state of the element follow design spec
- [X] selected state of the element follow design spec
- [X] selected hover state of the element follow design spec
- [X] pressed state of the element follow design spec
- [X] Responsiveness rules - the component has modifier classes for all breakpoints
- [X] Includes Compact/Cosy/Tablet design
- [X] RTL support
2. The code follows fundamental-styles code standards and style
- [X] only one top level `fd-*` class is used in the file
- [X] BEM naming convention is used
- [X] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [X] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [X] `fd-reset()` mixin is applied to all elements
- [X] Variables are used, if some value is used more than twice.
- [X] Checked if current components can be reused, instead of having new code.
3. Testing
- [X] tested Storybook examples with "CSS Resources" `normalize` option 
- [X] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [X] Verified all styles in IE11
- [X] Updated tests
- [X] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [X] Storybook documentation has been created/updated
- [X] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Review by N1XUS — APPROVED** [review]

File: src/table.scss [hunk 1]
```diff
@@ -598,6 +585,19 @@ $block: #{$fd-namespace}-table;
...
+    .#{$block}__body {
+      .#{$block}__cell {
+        &--checkbox {
+          position: relative;
+        }
+
+        .fd-checkbox + .fd-checkbox__label:empty {
+          top: -0.25rem;
+          position: absolute;
+        }
```
Comment (InnaAtanasova) [inline src/table.scss:598]: You can't modify properties of one component (checkbox in this case) inside another (table). This is violating the self-contained rules.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: src/table.scss (hunk 1 above)
Comment (platon-rov) [reply src/table.scss:598]: I agree, but in this case, we're adjusting it only for the checkboxes inside the table.

We also don't have any condensed mode for the checkbox, that's why I done it like that.

Comment (InnaAtanasova) [reply src/table.scss:598]: it doesn't matter that it's only for the checkboxes inside table. We have lots of cases like this. You can't modify props of checkbox inside table scss. You can always add additional styling inside checkbox or modifier class inside table. But this way we can't accept it.

**Review by InnaAtanasova — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
@platon-rov please update the wiki page. You are adding new classes that the implementation libraries need to add. Thanks

**Pull request merged**